qiskit/quantum_info/operators/symplectic/pauli.py

```diff
@@ -547,6 +547,20 @@ def anticommutes(self, other, qargs=None):
         """
         return np.logical_not(self.commutes(other, qargs=qargs))
 
+    def qw_anticommutes(self, other):
```

Comment:
also probably better to implement a `qubit_wise_commutes` method, then make this one a `np.logical_not` of that one. (just like it is done for the `commutes` and `anticommutes` methods above it)

Reply:
How about `Pauli.anticomutes(self, other, qubit_wise: bool=False)` like https://github.com/Qiskit/qiskit-terra/pull/5299?

Reply:
that's good too. 

I also am not sure about the definition of "anticommute" in this case. I know "qubit-wise commutation" is when all of the qubit operators commute with each other. But is "qubit-wise anticommutation" the logical-not of that? Or we need all qubit operators to anticommute with each other? These are not the same thing. Is there a definition of this somewhere? If not then I think we should just add the `qubit_wise` option to `Pauli.commutes()`.

Reply:
I implemented `qw_anticommutes` as a logical-not of Qubit-wise commuting (if there is at least one operator pair that anticommutes, then it is qubit-wise anticommuting). I'll add the `qubit_wise` option to Pauli.commutes() and then add the np.logical_not to `Pauli.anticommutes()` where the `qubit_wise` option is true.

Reply:
*  @irajput suggested adding `qubit_wise` flag to `commutes` before making this PR. I thought it was a good idea to hold off to see if there were a use case. It looks like a popular idea. I agree as well. 
* > I also am not sure about the definition of "anticommute" in this case
    
     I agree. qubit-wise commutation implies commutation. But, qubit-wise anti-commutation does not imply anti-commutation. I doubt there is an existing use case, and the concept doesn't seem useful in general. Unless there is something I don't see, I favor not implementing qubit-wise anticommutation at all.

Reply:
Yeah, I agree it might be better to find an alternative implementation to these functions that does not involve Qubit-wise anti commutation, so I'll work on that.

Reply:
hmm. `anti qubit-wise commuting` is correct, but it's confusing as you said. I think implementing only qubit_wise_commutes like `Pauli.commutes(self, other, qubit_wise: bool=False)` and use `not` to reverse the logic is easier to understand.

Reply:
> and use not to reverse the logic is easier to understand.


Yes, I think this is @irajput 's new approach.

Reply:
Actually, I just switched to the Numpy implementation which doesn't require any explicit commutes function.